File: app/src/main/java/com/anhhoang/popularmovies/model/Trailer.java
package com.anhhoang.popularmovies.model;
/*
 * Copyright (C) 2013 The Android Open Source Project
 */

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by AnhHo on 3/20/2017.
 */

public class Trailer implements Parcelable {
    private String id;
    private String iso6391;
    private String iso31161;
    private String key;
    private String name;
    private String site;
    private int size;
    private String type;

    public Trailer(Parcel source) {
        id = source.readString();
        iso6391 = source.readString();
        iso31161 = source.readString();
        key = source.readString();
        name = source.readString();
        site = source.readString();
        size = source.readInt();
        type = source.readString();
    }

    public String getId() {
        return id;
    }

    public String getIso6391() {
        return iso6391;
    }

    public String getIso31161() {
        return iso31161;
    }

    public String getKey() {
        return key;
    }

    public String getName() {
        return name;
    }

    public String getSite() {
        return site;
    }

    public int getSize() {
        return size;
    }

    public String getType() {
        return type;
    }

    @Override
    public int describeContents() {
        return hashCode();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(id);
        dest.writeString(iso6391);
        dest.writeString(iso31161);
        dest.writeString(key);
        dest.writeString(name);
        dest.writeString(site);
        dest.writeInt(size);
        dest.writeString(type);
    }

    public static final Parcelable.Creator<Trailer> CREATOR = new Creator<Trailer>() {
        @Override
        public Trailer createFromParcel(Parcel source) {
            return new Trailer(source);
        }

        @Override
        public Trailer[] newArray(int size) {
            return new Trailer[size];
        }
    };
}

File: MovieApp/app/src/main/java/com/anhhoang/popularmovies/model/Trailer.kt
package com.anhhoang.popularmovies.model

/*
 * Copyright (C) 2013 The Android Open Source Project
 */

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created by AnhHo on 3/20/2017.
 */

class Trailer(
        @SerializedName("id")
        var id: String,
        @SerializedName("iso6391")
        var iso6391: String,
        @SerializedName("iso31161")
        var iso31161: String,
        @SerializedName("key")
        var key: String,
        @SerializedName("name")
        var name: String,
        @SerializedName("site")
        var site: String,
        @SerializedName("size")
        var size: Int,
        @SerializedName("type")
        var type: String
) : Parcelable {
    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Trailer> = object : Parcelable.Creator<Trailer> {
            override fun createFromParcel(source: Parcel): Trailer = Trailer(source)
            override fun newArray(size: Int): Array<Trailer?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readInt(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(id)
        dest.writeString(iso6391)
        dest.writeString(iso31161)
        dest.writeString(key)
        dest.writeString(name)
        dest.writeString(site)
        dest.writeInt(size)
        dest.writeString(type)
    }
}
